app/models/manageiq/providers/amazon/storage_manager/s3/cloud_object_store_object.rb

```diff
@@ -18,6 +18,10 @@ def provider_object(connection = nil)
   end
 
   def raw_delete
-    with_provider_object(&:delete)
+    if key.end_with? "/" # delete object with subobjects (aka. folder)
```

Comment:
You are correct about this. When looking from the perspective of UI, it makes perfect sense. I was coming more from the provider perspective. For example, if this is called from the automate or miq API, what will the user expect - to delete the object itself (as S3 API does) or the entire hierarchy (as the UI does)?

That being said, I also don't mind merging this, but keep this in mind for future reference.